plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/decoder/KafkaRowDecoderFactory.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.kafka.decoder;
+
+import io.trino.decoder.DecoderColumnHandle;
+import io.trino.decoder.RowDecoder;
+import io.trino.decoder.RowDecoderFactory;
+import io.trino.decoder.avro.AvroDeserializer;
+import io.trino.decoder.avro.SingleValueRowDecoder;
+import io.trino.decoder.dummy.DummyRowDecoderFactory;
+import org.apache.avro.LogicalTypes.TimeMillis;
+import org.apache.avro.Schema;
+import org.apache.kafka.common.serialization.ByteBufferDeserializer;
+import org.apache.kafka.common.serialization.DoubleDeserializer;
+import org.apache.kafka.common.serialization.FloatDeserializer;
+import org.apache.kafka.common.serialization.IntegerDeserializer;
+import org.apache.kafka.common.serialization.LongDeserializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.trino.decoder.avro.AvroRowDecoderFactory.DATA_SCHEMA;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class KafkaRowDecoderFactory
```

Comment:
Can you please provide tests for that class? It looks unused in the commit which introduces it.